roles/openshift_master/tasks/bootstrap.yml

```diff
@@ -26,3 +26,66 @@
   copy:
     content: "{{ kubeconfig_out.stdout }}"
     dest: "{{ openshift_master_config_dir }}/bootstrap.kubeconfig"
+
+- name: create a temp dir for this work
+  command: mktemp -d /tmp/openshift_node_config-XXXXXX
+  register: mktempout
+  run_once: true
+
+# This generate is so that we do not have to maintain
+# our own copy of the template.  This is generated by
+# the product and the following settings will be
+# generated by the master
+- name: generate a node-config dynamically
+  command: >
+    {{ openshift_master_client_binary }} adm create-node-config
```

Comment:
I feel this is an unfinished feature in the product if we have to generate a config dynamically and then make a bunch of edits to it in the next task.  -1.

We should use a lookup(template) here, as it's quite large.

Reply:
We need a node config.  The reason I generated the config is that it maintains the settings for an out-of-the-box node.  Will that work? yes.  Is it what we want? no.  The following steps customize the config the way we want it.  I'd prefer to not manage the template if we don't have to.

Reply:
>  I thought the whole point of this effort was the master provides the configs. I don't see much value in the master providing configs that aren't what we want.
The master can provide us a config.  That config out of the box is a standard node-config.yaml.  That standard node-config.yaml is not exactly what we want it to be.

What the product currently delivers is a method to create a generalized node-config.yaml that is defined for a specific node type (master|infra|compute) and hand it back to us upon boot.

> I prefer we use a template and put in what we want, or master spits out what we want based on parameter we configure in the master. I would prefer we configure the master's config to contain the settings for a node's config versus having to make dynamic calls to the master.

I'm not against the master handling more config options for generating a node-config.yaml.  I'm fine with that approach.  Currently that does not exist.

I would prefer we not manage a template but only the specific settings that we want.  A dynamic node-config that is generated by the product is more likely up-to-date with the settings and features that are supported/needed vs us maintaining it separately.

Reply:
> What the product currently delivers is a method to create a generalized node-config.yaml that is defined for a specific node type (master|infra|compute) and hand it back to us upon boot.

It should hand us something we can use as-is.  That's whole point of getting the config from the master.

> I'm not against the master handling more config options for generating a node-config.yaml. I'm fine with that approach. Currently that does not exist.

I think this approach makes the most sense.  Configure you master, and the nodes come online as they're supposed to.  Having to bake machinations into the node to do these edits dynamically is confusing and going to be hard to maintain long-term.

> I would prefer that we not manage a template but only the specific settings that we want. A dynamic node-config that is generated by the product is more likely up-to-date with the settings and features that are supported/needed vs us maintaining it separately.
This is non-deterministic.  I want to know exactly what's in the config file if we're going to make edits to it.  I think this is a potential area of increased collaboration between us and the product owners; maintaining a proper config template should not be a huge undertaking.  We will eventually need to add migration/multiversion logic to a config file, it's inevitable.  That logic will either need to be carefully crafted through tasks, or it can be baked into a template.

Near-term making dynamic edits might be slightly better, long term I don't think it's the right direction.